metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/sensors/datahub_sensors.py

```diff
@@ -314,100 +315,177 @@ def process_asset_materialization(
-
-                # Emitting asset with upstreams and downstreams
-                dataset_urn = dagster_generator.emit_asset(
-                    self.graph,
-                    asset_key,
-                    materialization.description,
-                    properties,
-                    downstreams=downstreams,
-                    upstreams=upstreams,
-                    materialize_dependencies=self.config.materialize_dependencies,
-                )
+        if not asset_downstream_urn:
+            asset_downstream_urn = (
+                dagster_generator.asset_keys_to_dataset_urn_converter(asset_key)
+            )
+            context.log.info(
+                f"asset_downstream_urn from asset keys: {asset_downstream_urn}"
+            )
 
+        return asset_downstream_urn
+
+    def _process_lineage(
+        self,
+        context: RunStatusSensorContext,
+        dagster_generator: DagsterGenerator,
+        log: EventLogEntry,
+        asset_downstream_urn: DatasetUrn,
+    ) -> Tuple[Set[str], Set[str]]:
+        downstreams = {asset_downstream_urn.urn()}
+        upstreams: Set[str] = set()
+
+        if (
+            log.asset_materialization
+            and self.config.enable_asset_query_metadata_parsing
+        ):
+            try:
+                query_metadata = log.asset_materialization.metadata.get("Query")
+                if isinstance(query_metadata, TextMetadataValue):
+                    lineage = self.parse_sql(
+                        context=context,
+                        sql_query=str(query_metadata.text),
+                        env=asset_downstream_urn.env,
+                        platform=asset_downstream_urn.platform.replace(
+                            "urn:li:dataPlatform:", ""
+                        ),
+                    )
+                    if lineage and lineage.downstreams:
+                        if self.config.emit_queries:
+                            dagster_generator.gen_query_aspect(
+                                graph=self.graph,
+                                platform=asset_downstream_urn.platform,
+                                query_subject_urns=lineage.upstreams
+                                + lineage.downstreams,
+                                query=str(query_metadata.text),
+                            )
+                        downstreams = downstreams.union(set(lineage.downstreams))
+                        upstreams = upstreams.union(set(lineage.upstreams))
+                        context.log.info(
+                            f"Upstreams: {upstreams} Downstreams: {downstreams}"
+                        )
+                    else:
+                        context.log.info(f"Lineage not found for {query_metadata.text}")
+                else:
+                    context.log.info("Query not found in metadata")
+            except Exception as e:
+                context.log.error(f"Error in processing asset logs: {e}")
```

Comment:
is context.log a normal logging object? if so, you can use log.exception() or log.info(..., exc_info=True)